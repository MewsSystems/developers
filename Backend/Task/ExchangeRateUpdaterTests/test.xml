<?xml version="1.0" encoding="UTF-8"?>
<kurzy banka="CNB" datum="16.01.2020" poradi="11">
<tabulka typ="XML_TYP_CNB_KURZY_DEVIZOVEHO_TRHU">
<radek kod="AUD" mena="dolar" mnozstvi="1" kurz="15,622" zeme="Austrálie"/>
<radek kod="BRL" mena="real" mnozstvi="1" kurz="5,411" zeme="Brazílie"/>
<radek kod="BGN" mena="lev" mnozstvi="1" kurz="12,870" zeme="Bulharsko"/>
<radek kod="CNY" mena="žen-min-pi" mnozstvi="1" kurz="3,276" zeme="Čína"/>
<radek kod="DKK" mena="koruna" mnozstvi="1" kurz="3,368" zeme="Dánsko"/>
<radek kod="EUR" mena="euro" mnozstvi="1" kurz="25,170" zeme="EMU"/>
<radek kod="PHP" mena="peso" mnozstvi="100" kurz="44,330" zeme="Filipíny"/>
<radek kod="HKD" mena="dolar" mnozstvi="1" kurz="2,900" zeme="Hongkong"/>
<radek kod="HRK" mena="kuna" mnozstvi="1" kurz="3,380" zeme="Chorvatsko"/>
<radek kod="INR" mena="rupie" mnozstvi="100" kurz="31,779" zeme="Indie"/>
<radek kod="IDR" mena="rupie" mnozstvi="1000" kurz="1,654" zeme="Indonesie"/>
<radek kod="ISK" mena="koruna" mnozstvi="100" kurz="18,372" zeme="Island"/>
<radek kod="ILS" mena="nový šekel" mnozstvi="1" kurz="6,526" zeme="Izrael"/>
<radek kod="JPY" mena="jen" mnozstvi="100" kurz="20,497" zeme="Japonsko"/>
<radek kod="ZAR" mena="rand" mnozstvi="1" kurz="1,562" zeme="Jižní Afrika"/>
<radek kod="CAD" mena="dolar" mnozstvi="1" kurz="17,294" zeme="Kanada"/>
<radek kod="KRW" mena="won" mnozstvi="100" kurz="1,945" zeme="Korejská republika"/>
<radek kod="HUF" mena="forint" mnozstvi="100" kurz="7,540" zeme="Maďarsko"/>
<radek kod="MYR" mena="ringgit" mnozstvi="1" kurz="5,546" zeme="Malajsie"/>
<radek kod="MXN" mena="peso" mnozstvi="1" kurz="1,199" zeme="Mexiko"/>
<radek kod="XDR" mena="ZPČ" mnozstvi="1" kurz="31,135" zeme="MMF"/>
<radek kod="NOK" mena="koruna" mnozstvi="1" kurz="2,541" zeme="Norsko"/>
<radek kod="NZD" mena="dolar" mnozstvi="1" kurz="15,012" zeme="Nový Zéland"/>
<radek kod="PLN" mena="zlotý" mnozstvi="1" kurz="5,947" zeme="Polsko"/>
<radek kod="RON" mena="leu" mnozstvi="1" kurz="5,266" zeme="Rumunsko"/>
<radek kod="RUB" mena="rubl" mnozstvi="100" kurz="36,587" zeme="Rusko"/>
<radek kod="SGD" mena="dolar" mnozstvi="1" kurz="16,753" zeme="Singapur"/>
<radek kod="SEK" mena="koruna" mnozstvi="1" kurz="2,382" zeme="Švédsko"/>
<radek kod="CHF" mena="frank" mnozstvi="1" kurz="23,440" zeme="Švýcarsko"/>
<radek kod="THB" mena="baht" mnozstvi="100" kurz="74,222" zeme="Thajsko"/>
<radek kod="TRY" mena="lira" mnozstvi="1" kurz="3,856" zeme="Turecko"/>
<radek kod="USD" mena="dolar" mnozstvi="1" kurz="22,538" zeme="USA"/>
<radek kod="GBP" mena="libra" mnozstvi="1" kurz="29,448" zeme="Velká Británie"/>
</tabulka>
</kurzy>
