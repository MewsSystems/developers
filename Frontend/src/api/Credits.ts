import axios, { AxiosPromise } from 'axios'
import { RequestConfig } from './requestConfig'
import { Credits } from 'model/api/Credits'

export class CreditsApi {
  /**
   * Movie Credits
   *
   * @param movie_id
   */

  static getMovieCredits(
    movie_id: number,
    config?: RequestConfig
  ): AxiosPromise<Credits> {
    return axios({
      ...config,
      method: 'GET',
      url: `/movie/${movie_id}/credits`,
    })
  }
}
