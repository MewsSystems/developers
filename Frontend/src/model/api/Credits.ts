import { Cast } from './Cast'
import { Crew } from './Crew'

export interface Credits {
  id: number
  cast: Cast[]
  crew: Crew[]
}
