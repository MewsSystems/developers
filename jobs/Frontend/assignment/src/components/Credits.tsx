import { useEffect, useState } from "react";
import styled from "styled-components";
import { Avatar, Typography, WithMovieIdProps } from ".";
import { tmdbClient } from "@/tmdbClient";
import { Credits } from "tmdb-ts";

const CreditsWrapper = styled.div`
  display: flex;
  gap: 16px;
  width: 100%;

  overflow-x: auto;
`;

export function Credits({ movieId }: WithMovieIdProps) {
  const [data, setData] = useState<Credits>();

  useEffect(() => {
    tmdbClient.movies.credits(movieId).then(res => setData(res));
  }, [movieId]);

  return (
    <CreditsWrapper>
      {data?.cast.length || data?.crew.length ? (
        <>
          {data?.cast.map(({ name, profile_path, character, credit_id }) => (
            <Avatar key={credit_id} imgPath={profile_path} name={name} description={character} />
          ))}
          {data?.crew.map(({ name, profile_path, job, credit_id }) => (
            <Avatar key={credit_id} imgPath={profile_path} name={name} description={job} />
          ))}
        </>
      ) : (
        <Typography variant="titleMedium" color="secondary">
          No credits found
        </Typography>
      )}
    </CreditsWrapper>
  );
}
