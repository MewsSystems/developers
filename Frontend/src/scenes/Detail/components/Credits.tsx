import React from 'react'
import { useGetMovieCredits } from 'hooks/useGetMoveCredits'
import { useTranslation } from 'react-i18next'
import { Heading } from 'components/Heading'
import styled from 'styled-components'
import { CastCard } from 'components/CastCard'
import { useGetImagePath } from 'hooks/useGetImagePath'
import { ConfigurationProfileSizesEnum } from 'model/api/ConfigurationProfileSizes'
import { ErrorContent } from 'components/ErrorContent'
import { AnimatedLine } from 'components/Loading/AnimatedLine'

const Grid = styled.div`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(4, 1fr);
`

export interface CreditsProps {
  id: number
}

export const Credits: React.FC<CreditsProps> = ({ id }) => {
  const { t } = useTranslation('detail')
  const { credits, isLoading, hasErrored } = useGetMovieCredits(id)
  const getImagePath = useGetImagePath(ConfigurationProfileSizesEnum.W_185)

  if (hasErrored)
    return (
      <ErrorContent
        title={t('error-content.title')}
        text={t('error-content.text')}
      />
    )

  if (isLoading) return <AnimatedLine width="20rem" height="2rem" />

  if (credits) {
    const { cast, crew } = credits

    return (
      <div>
        <Heading level={2}>{t('cast')}</Heading>
        <Grid>
          {cast.map(({ id, profile_path, name, character }, index) => (
            <CastCard
              key={id + index}
              background={getImagePath(profile_path)}
              name={name}
              role={character}
            />
          ))}
        </Grid>
        <Heading level={2}>{t('crew')}</Heading>
        <Grid>
          {crew.map(({ id, profile_path, name, job }, index) => (
            <CastCard
              key={id + index}
              background={getImagePath(profile_path)}
              name={name}
              role={job}
            />
          ))}
        </Grid>
      </div>
    )
  }

  return null
}
