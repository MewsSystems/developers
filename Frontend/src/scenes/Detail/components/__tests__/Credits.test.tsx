import React from 'react'
import { shallowWithStore } from 'test/shallowWithStore'
import { Credits } from '../Credits'

describe('Test Credits scene', () => {
  it('Tests snapshot', () => {
    expect(shallowWithStore(<Credits id={1} />)).toMatchSnapshot()
  })
})
