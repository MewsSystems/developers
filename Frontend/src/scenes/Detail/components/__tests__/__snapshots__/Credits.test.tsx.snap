// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Credits scene Tests snapshot 1`] = `""`;
